src/typechecker.erl

```diff
@@ -1436,6 +1435,16 @@ glb_types({type, _, N, Args1}, {type, _, N, Args2}) ->
 glb_types(_, _) ->
     {type, 0, any, []}.
 
+get_rec_field_type(FieldName,
+                   [{typed_record_field,
+                     {record_field, _,
+                      {atom, _, FieldName}, _}, Ty}|_]) ->
+    Ty;
+get_rec_field_type(FieldName, [_|RecFieldTypes]) ->
+    get_rec_field_type(FieldName, RecFieldTypes);
+get_rec_field_type(_, []) ->
+    {error, not_found}.
```

Comment:
I would suggest throwing an exception here instead.

Reply:
👍 pushed a change to throw an exception.